### DISCUSSION TITLE
Can I specify special build processing for one .md file?

### DISCUSSION THREAD

**Discussion by ms2art** [squidfunk/mkdocs-material#4300] [General]
I have somehow arranged things so that one of my .md files gets built differently from all the others. The difference is that the right sidebar div (class=".md-sidebar md-sidebar--secondary") is not in the output HTML. That's exactly what I want for this particular page, but I can't remember how I did it. Someone else is taking over the doc builds, and I have no idea how to help them get the result I do. Any thoughts?

**Comment by ms2art** [comment]
I should have mentioned that I don't think it's the inclusion of a home.html in my theme_overrides folder. I do have a home.html in that folder, but it's a relic from an earlier project, and not used for my current project.

**Comment by squidfunk** [comment]
> Someone else is taking over the doc builds, and I have no idea how to help them get the result I do. Any thoughts?

I'm sorry, but I don't understand the question. It's pretty hard to help you when you can't explain what you're doing.

**Reply by ms2art** [reply]
Yeah, sorry. It's kind of confusing to me, too, and maybe not even appropriate for this channel. In case it helps: In my md files, there's one that I want to convert to HTML with the right sidebar div either hidden or entirely omitted. Currently, the div is in fact omitted from the output HTML for that one file. That's exactly what I want, but it only builds that way on my system, not on anyone else's. I must have done something to my environment to get the desired result, but I can't remember what. It isn't a case of me swapping in a static HTML file post-build; I've verified that the page is being built every time, along with all the other pages. It isn't a CSS trick, because the div is truly not included in the file in question. I don't see anything in my overrides folder that looks relevant. I guess the question is: Is there some method in the MkDocs/Material workflow for building a specific file with different HTML structure?

**Reply by squidfunk** [reply]
Yes, you can [provide a different template for each page](https://squidfunk.github.io/mkdocs-material/reference/#setting-the-page-template). Set up [theme extension](https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme) via `mkdocs.yml`:

``` yaml
theme:
  name: material
  custom_dir: overrides

```

Then create `overrides/custom.html` and [override any blocks](https://squidfunk.github.io/mkdocs-material/customization/#overriding-blocks) you want to change:

``` html
{% extends "base.html" %}

<!-- Your block overrides here -->
```

Then in your page:

``` md
---
template: custom.html
---

# Document title
...
```

**Reply by ms2art** [reply]
I'm not sure that can work. Using the content template only affects the center "md-content" div, and the site_nav template affects both the left and right sidebar divs. In my situation, the left sidebar is required (and its content may change in successive builds, as the TOC gets updated), so I need to keep that div in place. I don't want the right sidebar; it would be great if there was a separate template for that element.

**Reply by squidfunk** [reply]
Yes, it can work. You can override blocks or the entire template in `custom.html`, use custom partials, etc., i.e. provide something completely different than the stock templates. It has no limitations.

**Reply by ms2art** [reply]
Cool! Thank you; I will investigate further.
